hadoop-crypto/src/test/java/com/palantir/crypto2/hadoop/InMemoryKeyStorageStrategy.java

```diff
@@ -28,17 +30,21 @@ public final class InMemoryKeyStorageStrategy implements KeyStorageStrategy {
     public InMemoryKeyStorageStrategy() {}
 
     @Override
-    public void put(String fileKey, KeyMaterial keyMaterial) {
+    public void put(String fileKey, KeyMaterial keyMaterial) throws IOException {
         store.put(fileKey, keyMaterial);
     }
 
     @Override
-    public KeyMaterial get(String fileKey) {
+    public KeyMaterial get(String fileKey) throws IOException {
         return store.get(fileKey);
     }
 
     @Override
-    public void remove(String fileKey) {
+    public void remove(String fileKey) throws IOException {
+        if (!store.containsKey(fileKey)) {
+            // Mimic what the non-test-in-memory version would do if trying to remove it
+            throw new NoSuchFileException(fileKey);
```

Comment:
as mentioned above, I think it's probably worth explicitly returning the different error types rather than piggy backing on IOExceptions